ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/TransactionReceipt.java

```diff
@@ -154,8 +165,9 @@ private void writeTo(final RLPOutput out, final boolean withRevertReason) {
     out.writeLongScalar(cumulativeGasUsed);
     out.writeBytes(bloomFilter);
     out.writeList(logs, Log::writeTo);
-    if (withRevertReason && revertReason.isPresent()) {
-      out.writeBytes(revertReason.get());
+    if (withMetadata) {
+      out.writeBytes(revertReason.orElse(Bytes.EMPTY));
```

Comment:
I had a comment written out about using `out.writeNull` if not present but when I looked at RLPOutput the impl is just to write empty bytes.  Might be worth a comment pointing out that writing Bytes.EMPTY is the same as calling `RLPOutput.writeNull`